### PR TITLE
LDAP support and bug fixes.

### PR DESCRIPTION
**Pull request by msyk**
If you try to authenticate with LDAP server, see at the end of params.php file. There are some variables for LDAP communication. They are for Open Directory so far, I’m going to add other LDAP server such as Active Directory.

### CODE REVIEW

File: GenerateJSCode.php [hunk 1]
```diff
@@ -230,12 +230,14 @@ public function generateInitialJSCode($datasource, $options, $dbspecification, $
 
         $this->generateAssignJS(
             "INTERMediatorOnPage.isNativeAuth",
-            (isset($options['authentication']) && isset($options['authentication']['user'])
-                && ($options['authentication']['user'][0] === 'database_native')) ? "true" : "false");
+            ((isset($options['authentication'])
+                && isset($options['authentication']['user'])
+                && ($options['authentication']['user'][0] === 'database_native'))
+                || ((new LDAPAuth())->isActive)) ? "true" : "false");
```
Comment (matsuo) [inline GenerateJSCode.php]: Please add 
require_once(dirname(__FILE__) . '/LDAPAuth.php');
in GenerateJSCode.php. It seems to be needed.

File: DB_Proxy.php [hunk 2]
```diff
@@ -631,17 +631,17 @@ function processingRequest($options, $access = null, $bypassAuth = false)
...
-        } else if (file_exists($currentDirParam)) {
-            include($currentDirParam);
-        }
+//        $currentDirParam = $currentDir . 'params.php';
+//        $parentDirParam = dirname(dirname(__FILE__)) . DIRECTORY_SEPARATOR . 'params.php';
+//        if (file_exists($parentDirParam)) {
+//            include($parentDirParam);
+//        } else if (file_exists($currentDirParam)) {
+//            include($currentDirParam);
+//        }
```
Comment (matsuo) [inline DB_Proxy.php]: Please uncomment. Including params.php of the parent directory if exsits is needed.

File: GenerateJSCode.php (hunk 1 above)
Comment (msyk) [inline GenerateJSCode.php]: I think the autoloader is working appropriately without require_onece statement.

File: DB_Proxy.php (hunk 2 above)
Comment (msyk) [inline DB_Proxy.php]: I think any variables come from params.php don't exist in the processingRequest method any more.

Comment (matsuo) [inline DB_Proxy.php]: I see. Exactly.

File: GenerateJSCode.php (hunk 1 above)
Comment (matsuo) [inline GenerateJSCode.php]: How about PHP 5.2? Test of Travis CI failed.

Comment (msyk) [inline GenerateJSCode.php]: IM's class autoloader depends on the spl_autoload_register function. It's supported from 5.1.2. Doesn't it exist the SPL function support?

Comment (msyk) [inline GenerateJSCode.php]: As far as TravisCI's results, PHP ver.5.3 reports the same failure of the test case. Is that a kind of unstable situation of Travis CI?

Comment (msyk) [inline GenerateJSCode.php]: Anyway, I added reqiure_once on the next commit.

Comment (msyk) [inline GenerateJSCode.php]: Oops! This issue is described below. new xx -> isn't work under 5.3.
http://stackoverflow.com/questions/13388541/php-parse-error-syntax-error-unexpected-t-object-operator

Comment (msyk) [inline GenerateJSCode.php]: bingo!

Comment (matsuo) [inline GenerateJSCode.php]: Oh, I see! Thanks for fixing.

**Pull request merged**